x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -211,7 +214,8 @@ public void execute(
                         computeListener.acquireCompute().delegateFailure((l, profiles) -> {
```

Comment:
Again, here I am not 100% sure of the intended logic - for a remote, it looks like if `configuration.allowPartialResults()` is true and any result is delivered, any error is converted to partial. Is that also the case for local results? If yes, I don't see what checks it in the code. If no, then it looks like local and remote behaves differently with regard to partial failure logic?

Reply:
Yes, the search_shards can fail when a node shuts down, and I've pushed https://github.com/elastic/elasticsearch/pull/122708/commits/d3be47f51828e6d03ae3951434c37b8535cee479 to handle this case. We also need to handle partial results for the final pipeline, but that's for a follow-up.